### PR TITLE
fix(service-portal): Updating occupation licenses

### PR DESCRIPTION
**Pull request by jonbjarnio**
## What

- Updated health directorate api
- Adjusted UI to changes
- Removed old feature flag from health
- Improved logging in occupational license service
- Refactor graphql layer / ui, easier implementation for future licenses

image

image

image

image

## Why

Bug fixes and improvements 

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts [hunk 1]
```diff
@@ -32,18 +40,18 @@ export abstract class OccupationalLicense {
...
   profession!: string
 
-  @Field(() => Boolean)
-  isValid!: boolean
-
   @Field(() => String)
   validFrom!: string
+
+  @Field(() => String)
+  isValid!: Validity
```
Comment (thorkellmani) [inline libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts:54]:
```suggestion

  @Field(() => Validty)
  isValid!: Validity
```

Why not make a graphql enum `Validity` (with registerEnumType) and typecheck it here so you don't need to do the `license.isValid as Validity`

**Review by thorkellmani — CHANGES_REQUESTED** [review]

File: libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts [hunk 2]
```diff
@@ -11,6 +11,14 @@ export enum OccupationalLicenseType {
   HEALTH = 'HEALTH_DIRECTORATE',
 }
 
+export enum HEATLH_DIRECTORATE_STATUS_TYPE {
+  valid = 'Í gildi',
+  limited = 'Í gildi - Takmörkun',
+  error = 'Ógilt',
```
[the remark this answers is not in the document]
Comment (jonbjarnio) [reply libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts:17]: This will not be displayed to the user, those are the values that health directorate api status property. Only used to change UI colors and messages displayed to users.

File: libs/service-portal/health/src/screens/Dentists/Dentists.tsx [hunk 3]
```diff
@@ -1,7 +1,5 @@
-import { useEffect, useState } from 'react'
+import { useState } from 'react'
```
Comment (osk) [inline libs/service-portal/health/src/screens/Dentists/Dentists.tsx:1]: This should be in a separate PR.

File: libs/service-portal/occupational-licenses/src/components/LicenceActionCard.tsx [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
 import { m } from '@island.is/service-portal/core'
+import { OccupationalLicenseStatus } from '@island.is/api/schema'
 
 type LicenseActionCardProps = {
   type?: string
   validFrom?: string | Date
   url?: string
   image?: string
-  isValid: boolean
+  isValid: OccupationalLicenseStatus
```
Comment (osk) [inline libs/service-portal/occupational-licenses/src/components/LicenceActionCard.tsx:12]:
```suggestion
  status: OccupationalLicenseStatus
```

Since this is now an enum `isValid` is not descriptive

File: libs/service-portal/occupational-licenses/src/components/LicenseDetail.tsx [hunk 5]
```diff
@@ -13,7 +14,7 @@ type LicenseDetailProps = {
   licenseType?: string | null
   publisher?: string | null
   dateOfIssue?: string | null
-  isValid?: boolean
+  isValid?: OccupationalLicenseStatus
```
Comment (osk) [inline libs/service-portal/occupational-licenses/src/components/LicenseDetail.tsx:17]:
```suggestion
  status?: OccupationalLicenseStatus
```

File: libs/service-portal/occupational-licenses/src/components/LicenseDetail.tsx [hunk 6]
```diff
@@ -107,11 +108,29 @@ export const LicenseDetail: React.FC<LicenseDetailProps> = ({
...
+                      : isValid === 'limited'
+                      ? 'warning'
+                      : 'closeCircle'
+                  }
+                  color={
+                    isValid === 'valid'
+                      ? 'mint600'
+                      : isValid === 'limited'
+                      ? 'yellow600'
+                      : 'red600'
```
Comment (osk) [inline libs/service-portal/occupational-licenses/src/components/LicenseDetail.tsx:132]: Not consistent with `LicenceActionCard.tsx` that has

```
        variant:
          isValid === 'valid'
            ? 'blue'
            : isValid === 'limited'
            ? 'yellow'
            : 'rose',
``` 

is that on purpose?

File: libs/service-portal/health/src/module.tsx [hunk 7]
```diff
@@ -64,14 +64,12 @@ export const healthModule: PortalModule = {
     {
       name: hm.healthCenterRegistrationTitle,
       path: HealthPaths.HealthCenterRegistration,
-      key: 'HealthTransfer',
```
Comment (osk) [inline libs/service-portal/health/src/module.tsx:67]: separate PR

File: libs/api/domains/rights-portal/src/lib/rightsPortal.resolver.ts [hunk 8]
```diff
@@ -156,7 +156,6 @@ export class RightsPortalResolver {
     return this.rightsPortalService.getDentistStatus(user)
   }
 
-  @FeatureFlag(Features.servicePortalTransferHealthCenter)
```
Comment (osk) [inline libs/api/domains/rights-portal/src/lib/rightsPortal.resolver.ts:159]: separate PR

**Review by osk — CHANGES_REQUESTED** [review]

File: apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
+import { BrowserContext, expect, test } from '@playwright/test'
+import { icelandicAndNoPopupUrl, urls } from '../../../../support/urls'
+import { session } from '../../../../support/session'
+import { label } from '../../../../support/i18n'
+import { m } from '@island.is/service-portal/core/messages'
+import { disableI18n } from '../../../../support/disablers'
+
+const homeUrl = `${urls.islandisBaseUrl}/minarsidur`
+test.use({ baseURL: urls.islandisBaseUrl })
```
Comment (AndesKrrrrrrrrrrr) [inline apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts:9]: Probably not needed since you're just testing the service portal.

File: apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        await page.goto(icelandicAndNoPopupUrl('/minarsidur/starfsleyfi'))
+
+        const hasHealthDirectorateLicense = page
+          .locator(`role=button[name="${label(m.view)}"]`)
+          .first()
+
+        // Act
+        await hasHealthDirectorateLicense.click()
+        const healthDirectorateTitle = page.getByText('Sálfræðingur')
+        const healthDirectorateIsValid = page.getByText('Í gildi')
```
Comment (AndesKrrrrrrrrrrr) [inline apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts:45]: Why not use `label(...)`?

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]
Minor comments, otherwise LGTM (QA)

File: apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts (hunk 10 above)
Comment (jonbjarnio) [reply apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts:45]: Not an experienced tester, whats the difference if you care to elaborate?

Comment (jonbjarnio) [reply apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts:45]: The reason I used `getByText` is because the value is coming from an API reponse and is not translated.

Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts:45]: I was suggesting `label` to get translations and prevent future breakage when the text changes in code but not here.

If it's not in our strings library, should it maybe be?

File: apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts (hunk 9 above)
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/src/tests/islandis/service-portal/smoke/occupational-licenses.spec.ts:9]: `test.use` sets the base url you'll be using. However, it's already set in the base config. I.e. `test.use({baseURL:...})` is only useful if your app uses a sub-domain e.g. https://loftbru.island.is

File: libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts [hunk 11]
```diff
@@ -5,16 +5,37 @@ import {
...
+}
+
+export enum OccupationalLicenseStatus {
+  valid = 'valid',
+  error = 'error',
+  limited = 'limited',
+}
+
```
Comment (thorkellmani) [inline libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts:26]: This is almost the same thing, might be worth mapping the `healthDirectorateStatustType` to `OccupationalLicenseStatus`maybe in the Occupational-Licenses domain to make it more generic

**Review by thorkellmani — APPROVED** [review]
Approved,  but you really should look into splitting the `occupationalLicensesService.ts` up

File: libs/api/domains/occupational-licenses/src/lib/occupationalLicenses.service.ts [hunk 12]
```diff
@@ -26,149 +70,200 @@ export class OccupationalLicensesService {
...
-        []
+      const licenses = await this.healthDirectorateApi
+        .getHealthDirectorateLicense(user)
+        .catch((e) => {
+          if (e.status === 404) {
+            return []
+          }
+          throw new Error('Error getting health directorate license by id')
+        })
+        .then((licenses) => licenses ?? [])
```
[the remark this answers is not in the document]
Comment (jonbjarnio) [reply libs/api/domains/occupational-licenses/src/lib/occupationalLicenses.service.ts:83]: It is because the API responds with 404 error if there does not exists a license for that user. I do not want to log an error, when there is no license.

File: libs/api/domains/occupational-licenses/src/lib/occupationalLicenses.service.ts [hunk 13]
```diff
@@ -26,149 +70,200 @@ export class OccupationalLicensesService {
...
-
-      const today = new Date()
+      const licenses = await this.healthDirectorateApi
+        .getHealthDirectorateLicense(user)
+        .catch((e) => {
+          if (e.status === 404) {
+            return []
+          }
+          throw new Error('Error getting health directorate licenses')
+        })
```
[the remark this answers is not in the document]
Comment (jonbjarnio) [reply libs/api/domains/occupational-licenses/src/lib/occupationalLicenses.service.ts:146]: Same as above

File: libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts [hunk 14]
```diff
@@ -32,25 +53,25 @@ export abstract class OccupationalLicense {
   institution!: OccupationalLicenseType
 
   @Field(() => ID)
-  id!: string
+  id!: string | number
```
[the remark this answers is not in the document]
Comment (jonbjarnio) [reply libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts:56]: I did this because the id types depend on the licence, one license has number while other has guid string.

**Review by osk — APPROVED** [review]
